examples/assembly/Add_Array_(AArch64).asm

```diff
@@ -0,0 +1,191 @@
+# This file demonstrates adding 2 arrays using serial, SVE and NEON instructions.
+# Compile with aarch64-linux-gnu-gcc -mcpu=generic+sve (or -march=armv8.2-a+sve),
+# the compiler driver sets up code to call main.
+
+# All functions have the following prototype:
+# void addarrays_{serial,neon,sve} (int *restrict res, int *A, int *B, long N)
+# x0 -> res
+# x1 -> A
+# x2 -> B
+# x3 -> N
+
+.globl addarrays_serial
+.type addarrays_serial, %function
+
+addarrays_serial:
+    mov x4, xzr
+    b .cond
+.loop_body:
+    ldr w5, [x1, x4, lsl 2]
+    ldr w6, [x2, x4, lsl 2]
+    add w5, w5, w6
+    str w5, [x0, x4, lsl 2]
+    add x4, x4, 1
+.cond:
+    cmp x4, x3
+    blt .loop_body
+    ret
+
+.global addarrays_sve
+.type addarrays_sve, %function
+
+# We start by building predicate in p0 whose elements are set to 1,
+# for values of induction variable < N.
+# Then we load elements from input arrays x1, x2 into SVE registers z0, z1
+# for elements corresponding to active lanes of p0 (set to 1), 
+# and zero out the rest (/z).
+# The result is then added and computed in z0, and only elements from z0
+# corresponding to active lanes of p0 are stored into res.
+# Finally, the induction variable x4 is incremented by number of 32-bit words
+# in the vector.
+# If none of the predicate elements are active, the loop is terminated.
+#
+# Note that the code doesn't rely on knowing vector length, and can
+# thus "scale" to different vector lengths set by hardware without
+# needing recompilation, and also doesn't need a tail scalar loop.
+#
+# Eg: With vector length = 256 bits, ie 8 number of 32-bit elements.
```

Comment:
`# For example: With vector length = 256 bits, which is 8 of 32-bit elements.'

Avoiding Eg and ie makes this more straightforward to read.

Reply:
Also you should say what our total elements here is, seems like it's 10? 8, then 2, then exit?